test/rematch.jl

```diff
@@ -47,6 +47,190 @@ assertion_error = (VERSION >= v"0.7.0-DEV") ? LoadError : AssertionError
     end
 end
 
+@testset "Match using extractors" begin
+    function sub1(x) x+1 end
+
+    # sub1(4) == 3
+    let x = nothing
+        @test (@match 3 begin
+           sub1(x) => x
+        end) == 4
```

Comment:
I don't understand why this returns `4`. I could understand if it returned `2` or `3`, but i don't quite understand how it returns `4`.

The definition of `sub1(x)` is `sub1(x) = x+1`. So I take `@match 3 begin   sub1(x) => x  end` to mean "if `3` matches `sub1(x)` for some `x`, return that `x`." And in this case, `3` would match `sub1(2)`, so i'd expect it to return `2`?

Also, i'm confused why `x+1` is called `sub1` - i'd have called it `add1`?

Reply:
Oops sorry i left this comment on an outdated commit. Lemme copy it to the latest commit.